Scripts to run skipfile tests, push/pull results to SQUAD

katieworton opened the pull request:
Patches with scripts to run generated skipfile tests and push the results to SQUAD. Builds on https://github.com/Linaro/squad-client-utils/pull/22

katieworton committed:
run-and-push-skipfile-results.sh: Skipfile run/push results

Scripts for running the skipfile reproducers and pushing the results to
SQUAD.

Signed-off-by: Katie Worton <katie.worton@linaro.org>

roxell commented on generate-skipfile-plan-names:
```diff
@@ -0,0 +1,48 @@
+linux_mainline_next_branches = {
+    "linux-mainline": "master",
+    "linux-next": "master",
+}
```
This should not be here... we should be pulled from SQUAD.
However, before that just put them in the lib first.

katieworton replied:
Opened https://github.com/Linaro/squad-client-utils/pull/28 for feedback on draft code for auto detecting branches.

roxell commented:
$ ./generate-skipfile-plan-names
skipfile-reproducer-linux-stable-rc-linux-4.14.y-plan.yaml
skipfile-reproducer-linux-stable-rc-linux-4.19.y-plan.yaml
skipfile-reproducer-linux-stable-rc-linux-5.4.y-plan.yaml
skipfile-reproducer-linux-stable-rc-linux-5.10.y-plan.yaml
skipfile-reproducer-linux-stable-rc-linux-5.15.y-plan.yaml
skipfile-reproducer-linux-stable-rc-linux-6.1.y-plan.yaml
skipfile-reproducer-linux-stable-rc-linux-6.2.y-plan.yaml
skipfile-reproducer-linux-mainline-master-plan.yaml
skipfile-reproducer-linux-next-master-plan.yaml

but the files are empty

katieworton commented:
> $ ./generate-skipfile-plan-names skipfile-reproducer-linux-stable-rc-linux-4.14.y-plan.yaml skipfile-reproducer-linux-stable-rc-linux-4.19.y-plan.yaml skipfile-reproducer-linux-stable-rc-linux-5.4.y-plan.yaml skipfile-reproducer-linux-stable-rc-linux-5.10.y-plan.yaml skipfile-reproducer-linux-stable-rc-linux-5.15.y-plan.yaml skipfile-reproducer-linux-stable-rc-linux-6.1.y-plan.yaml skipfile-reproducer-linux-stable-rc-linux-6.2.y-plan.yaml skipfile-reproducer-linux-mainline-master-plan.yaml skipfile-reproducer-linux-next-master-plan.yaml
> 
> but the files are empty

Currently this is designed to generate the possible plan names, then the run script should check if each one exists before submitting it. This isn't great as it means this file name generator has to be kept aligned with the skipfile plan generation code.

Would it be better to have a library function that generates the names for the plans (that both this script and the skipfile script can use)?

katieworton commented:
@roxell - if you could look over my comments here that would be great :)

roxell committed:
squadutilslib: add get_projects() functions

Add function get_projects(), to get a list of projects.

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>
Signed-off-by: Katie Worton <katie.worton@linaro.org>

katieworton committed:
squad-create-skipfile-reproducers: Reorder arguments

Reorder arguments in squad-create-skipfile-reproducers so they appear in
alphabetical order and update the README to reflect this reorder.

Signed-off-by: Katie Worton <katie.worton@linaro.org>

katieworton committed:
squad-create-skipfile-reproducers: Fix project fetching

Update the skipfile reproducer generation script to use the utilities
for automatically fetching projects, rather than maintaining a list of
projects.
Update the README to reflect these changes.

Signed-off-by: Katie Worton <katie.worton@linaro.org>

katieworton commented:
Updated code to use automatic fetching of projects and pattern match plan file names rather than hard coding them.

Ready for review @roxell

katieworton committed:
Reproducer/skipfile scripts: change python shebang

Change Python shebangs from `/usr/bin/python3` to `/usr/bin/env python3`
for skipfile-related scripts for improved compatibility with virtual
environments.

Signed-off-by: Katie Worton <katie.worton@linaro.org>

katieworton committed:
requirements.txt: Add requirements

Add a requirements file and update README to show install steps.

Signed-off-by: Katie Worton <katie.worton@linaro.org>

katieworton commented:
- Added sign-off to updated get_projects patch
- Updated `./run-and-push-skipfile-results` to `./run-and-push-skipfile-results.sh`
- Updated commits to contain `<filename>: ...`
- Added a `requirements.txt`
- Changed `#!/usr/bin/python3` to `#!/usr/bin/env python3` for skipfile-related files

@roxell - changes ready for review
